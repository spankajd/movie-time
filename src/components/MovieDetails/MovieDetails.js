import React from 'react';

import { useSelector } from 'react-redux';

import { integerToRoman } from "../../utils/helper";

import './MovieDetails.scss';
import { selectSelectedMovie } from '../../store/selectors/movieSelectors';
import Rating from '../Rating/Rating';

const MovieDetails = () => {
    const selectedMovie = useSelector(selectSelectedMovie);

    return <div className="movie-details">
        { selectedMovie ? <>
            <div className="movie-title">Episode {integerToRoman(selectedMovie?.episode_id)} - {selectedMovie?.title}</div>
            <div className='movie-section'>
                <img className='movie-poster' src={selectedMovie?.Poster}/>
                
                <ul className='movie-properties'>
                    <li>
                        <strong>Director:</strong> { selectedMovie?.Director}
                    </li>
                    <li>
                        <strong>Language:</strong> { selectedMovie?.Language}
                    </li>
                    <li>
                        <strong>Genre:</strong> { selectedMovie?.Genre}
                    </li>
                    <li>
                        <strong>Country:</strong> { selectedMovie?.Country}
                    </li>
                    <li>
                        <strong>Average rating:</strong> <Rating rating={parseInt(selectedMovie?.imdbRating)} />
                    </li>
                </ul>
                
            </div>
            <div className='movie-footer'>
                <p>
                    {selectedMovie?.opening_crawl}
                </p>
            </div>
            </> : <div className='message'>
                Please select movie from the list
            </div> }
    </div>
}

export default MovieDetails;