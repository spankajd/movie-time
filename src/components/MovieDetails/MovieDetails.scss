.movie-details {
 display: flex;
 width: calc(40% - 30px);
 padding: 15px;
 flex-direction: column;

 .movie-title {
    font-size: 20px;
    margin-bottom: 10px;
 }
 .movie-section {
    display: inline-flex;
    flex-direction: row;
    gap: 10px; 

    p {
        margin: 0;
    }
 }
 .movie-poster {
    height: 100%;
    max-height: 200px;
    width: 40%;
 }
 .movie-properties {
    list-style: none;
    padding: 0;
    margin: 0;
    width: 60%;
    li {
        &:first-child {
            margin-top: 0;
        }
        margin: 10px 0;
    }
 }


 @media (max-width: 768px) {
    width: calc(100% - 20px);
    border-bottom: 1px solid #8badce;
    padding: 10px;
    .movie-title {
        display: none;
    }
 }
 
}