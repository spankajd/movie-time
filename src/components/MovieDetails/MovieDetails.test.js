import { screen } from "@testing-library/react";
import '@testing-library/jest-dom'; // Provides additional matchers
import MovieDetails from "./MovieDetails";
import { renderWithProviders } from "../../utils/test-utlis";

describe("MovieDetails Component", () => {
    it("renders MovieDetails", () => {
        renderWithProviders(
            <MovieDetails />
        );
        const element = screen.getByText(/Please select movie from the list/i);
        expect(element).toBeInTheDocument();
    });
    
    it("matches snapshot", () => {
        const { asFragment } = renderWithProviders(<MovieDetails />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders MovieDetails with selectedMovie", () => {
        renderWithProviders(
            <MovieDetails />, {
                preloadedState :{
                    moviesState : {
                        selectedMovie: {
                            title:'Test'
                        }
                    }
                }
                
            }
        );
        const element = screen.getByText(/Test/i);
        expect(element).toBeInTheDocument();
    });
});