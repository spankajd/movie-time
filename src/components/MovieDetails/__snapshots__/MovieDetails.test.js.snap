// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="movie-details"
  >
    <div
      class="message"
    >
      Please select movie from the list
    </div>
  </div>
</DocumentFragment>
`;
